File: Python/Easy/counting_bits.py
def count(n):
    list = []
    for i in range(n+1):
        t = []
        while i > 0:
            rest = i % 2      # converting a number to the binary system without bin
            t.append(rest)    # by saving the remainder of division by two
            i //= 2
        if sum(t) > 1:
            list.append(sum(t))
        else:
            list.append(sum(t))
    return list

File: C++/Easy/counting_bits.cpp
#include <bits/stdc++.h>
#include <numeric>
using namespace std;

vector<int> count(int n) {
    
    vector<int> list;

    for (int i = 0; i <= n; i++) {
        int x = i;
        int s = 0;

        while (x > 0) {
            int rest = x % 2;
            s += rest;
            x /= 2;
        }

        list.push_back(s);
    }

    return list;
}